### PR TITLE
More README and a few new things

### PR DESCRIPTION
**Pull request by qrush**
Hi there! I've been looking for such a useful tool for a while. Heck I'd pay for this (or maybe charge others). In the meantime, here's:

1. More README help
2. Introducing axios-debug-log since I wasn't sure why things weren't importing for this month
3. Bumped up `limit` on the Splitwise `/get_expenses` 
4. Running the event handler immediately since I got tired of waiting 1 minute to see anything happen
5. Take the date from the Splitwise expense for the YNAB transaction directly
6. New config value for `START_DATE` (I was trying to pull in this entire month's transactions)

This is a lot of stuff in one PR! If it's hard to follow let me know.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,46 @@ YNAB_ACCOUNT_ID - your YNAB account ID
 YNAB_PAYEE_ID - the default Payee ID you want to assign to your transactions
 YNAB_API_KEY - YNAB API key, generated on account's developer settings
 YNAB_UNCATEGORIZED_ID - YNAB "Needs categorization" ID, it's a master category that you can retrieve via YNAB's API, not sure if ID is general or per user.
+
+CRON_FREQUENCY - how often to run the sync if you're deploying it to Heroku
```
Comment (vascopinho) [inline README.md]:
```suggestion
CRON_FREQUENCY - how often to run the sync
```

File: src/client/SplitwiseClient.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import Axios, { AxiosInstance } from "axios";
 import { appConfig } from "../config/AppConfig";
 import { ExpensesResponse } from "../types/Splitwise";
+import "axios-debug-log";
```
Comment (vascopinho) [inline src/client/SplitwiseClient.ts]: I haven't checked but would be useful if we could reuse winston for logging with the http interceptor somehow :)
also for the sake of consistence logging format

File: src/config/AppConfig.ts [hunk 3]
```diff
@@ -15,4 +15,5 @@ export const appConfig: AppConfig = {
     userId: process.env.SPLITWISE_USER_ID,
   },
   CRON_FREQUENCY: process.env.CRON_FREQUENCY || "15",
+  START_DATE: process.env.START_DATE ? new Date(Date.parse(process.env.START_DATE)) : new Date(),
```
Comment (vascopinho) [inline src/config/AppConfig.ts]: thanks for this, obviously wiser this way

File: src/services/EventHandler.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ export class EventHandler {
 
   public async runHandler() {
     const result = await this.splitwiseClient.getExpenses();
+    logger.info(`Found ${result.expenses.length} expenses!`);
```
Comment (vascopinho) [inline src/services/EventHandler.ts]: Don't really see the point on this logging line. As we are getting all user expenses on Splitwise, this includes payments, deleted, settled since the beggining of time for the user :) this means that I get 100 transaction out of which none if going to be processed

**Review by vascopinho — COMMENTED** [review]
Thanks a lot for you feedback and contribution 🥇 
Left some comments in case you want to have a look at them! In any case, they are obviously improvements :)

File: src/client/SplitwiseClient.ts (hunk 2 above)
Comment (qrush) [reply src/client/SplitwiseClient.ts]: I think it'll be easier to just take this out for now - this was more for my personal poking around to see how this worked.

**Comment by qrush** [comment]
@vascopinho took out some of the logging in 7b8e663, thanks for the feedback.

**Comment by vascopinho** [comment]
I think we should be good for now, let me know what you think of the generic http client and axios interceptors

**Comment by qrush** [comment]
👍 Big fan, loving it! More HTTP logs, the better.

I think my main concern is filling in the "Payee" for each of my transactions that were imported - I think YNAB will eventually learn these based on the memo. Even so this still saves a ton of time.

Screen Shot 2020-11-28 at 13 02 05

I think the next steps are wiring up a Deploy to Heroku button and more docs for anyone else who'd like to use this.
